content/blog/2024/anchor-position-polyfill-2.md

```diff
@@ -0,0 +1,119 @@
+`::after` pseudo-elements as anchoring elements.
+
+{% callout 'note', false %}
+**What pseudo-elements can be anchors?**
+
+The [specification](https://drafts.csswg.org/css-anchor-position-1/#target)
+defining which pseudo-elements can be anchors was recently updated from
+`part-like` to `fully styleable tree-abiding pseudo-element`. This means that
+`::file-selector-button` and `::details-content` are not anchors.
+`::-webkit-slider-thumb` is not likely to fit this criteria either, which is a
+shame as it's been [extremely](https://codepen.io/jamessw/pen/KKLMJKm)
+[useful](https://codepen.io/jamessw/pen/ZENpWao) for examples.
+
+{% endcallout %}
+
+## anchor-scope support
+
+[@mmalerba](https://github.com/mmalerba) added support for `anchor-scope`, which
+is extremely useful when working with lists of elements that each have their own
+anchor elements.
+
+In this example, anchoring is applied with shared `.target` and `.anchor` rules,
+meaning there are multiple elements with `anchor-name: --list-item` set. By
+default, the last element with that rule will be the anchor.
+
+But that's not what we want. We want the anchoring element within the same list
+item to be our anchor. To do that, set `anchor-scope: --list-item` on
+the `li` element, to limit the search for an anchoring element.
+
+{{ embed.codepen(
+  id='qBeJePX',
+  title='Popover with anchor-scope',
+  user='jamessw'
+) }}
+
+Note that Chromium released support for `anchor-scope` in version 131, but since
+the polyfill doesn't support partial polyfilling, this demo will only work in
+Chromium before 125, after 131, or in a non-Chromium browser.
+
+## Position fallbacks
+
+What happens when an anchored element doesn't fit where it would go? We
+implemented the ["Overflow
+Management"](https://drafts.csswg.org/css-anchor-position/#fallback) portion of
+the spec to handle that.
+
+In simple cases, you could add `position-try: flip-block, flip-inline,
+flip-start;` to first try flipping over the block axis, then the inline axis,
+then over a diagonal axis. You can even combine the options.
+
+{{ embed.codepen(
+  id='abrZPQY',
+  title='Fallbacks with `position-try`',
+  user='jamessw'
+) }}
+
+## Performance improvements
+
+v0.3.0 was a performance-focused release. It turns out that applying anchor
+positioning is a fairly intensive process, which is one of the reasons why it's
+getting moved from the realm of JavaScript into the native browser.
+
+[@marchbox](https://github.com/marchbox) contributed a huge win that allows you
+to apply the polyfill to just the css files that impact anchor positioning. By
+default, the polyfill parses all of your CSS, which is likely more than you
+need. You're now able to specify exactly which CSS stylesheets contain rules
+that impact how you use anchor positioning, so not all CSS has to be parsed.
+
+Native implementations are able to calculate anchor positioning much more efficiently.
```

Comment:
I'd probably flesh this out a bit or remove it? It seems related, but a bit tangential.

```suggestion
```